sklearn/cluster/optics_.py

```diff
@@ -580,13 +586,10 @@ def _automatic_cluster(reachability_plot, ordering,
     """
 
     min_neighborhood_size = 2
-    min_cluster_size = int(min_cluster_size_ratio * len(ordering))
+    if min_cluster_size <= 1:
+        min_cluster_size = max(2, min_cluster_size * len(ordering))
     neighborhood_size = int(min_maxima_ratio * len(ordering))
 
-    # Should this check for < min_samples? Should this be public?
```

Comment:
Well, I'm changing 5 to 2 above (using `max` instead of `if`). @adrinjalali and I seem to agree that `min_samples` is not the relevant criterion here.

Reply:
The role of `min_samples` is kinda different in `OPTICS` than `DBSCAN`. In OPTICS you can have the generator epsilon set to a large number (it's `inf` by default), to generate a smooth `reachability_` array. Similarly, `min_samples` can also be set relatively large. Still, since the clusters are detected hierarchically, it can be the case that a cluster is smaller than the value for `min_samples`, and that's fine. I agree that it doesn't make much sense probably to do so, but they're independent parameters IMO.